Submitted By:            Igor Živković <contact@igor-zivkovic.from.hr>
Date:                    2013-09-07
Initial Package Version: 2.82
Upstream Status:         Unknown
Origin:                  OpenBSD
Description:             Allows compilation of Qt GUI client with Qt4

diff -Naur transmission-2.82.orig/qt/favicon.cc transmission-2.82/qt/favicon.cc
--- transmission-2.82.orig/qt/favicon.cc	2013-08-09 04:45:47.410416327 +0200
+++ transmission-2.82/qt/favicon.cc	2013-09-07 15:14:48.064094302 +0200
@@ -14,7 +14,12 @@
 #include <QNetworkAccessManager>
 #include <QNetworkReply>
 #include <QNetworkRequest>
-#include <QStandardPaths>
+
+#if QT_VERSION >= 0x050000
+# include <QStandardPaths>
+#else
+# include <QDesktopServices>
+#endif
 
 #include "favicon.h"
 
@@ -40,7 +45,11 @@
 QString
 Favicons :: getCacheDir( )
 {
+#if QT_VERSION >= 0x050000
     const QString base = QStandardPaths::writableLocation (QStandardPaths::CacheLocation);
+#else
+    const QString base = QDesktopServices::storageLocation( QDesktopServices::CacheLocation );
+#endif
     return QDir( base ).absoluteFilePath( "favicons" );
 }
 
diff -Naur transmission-2.82.orig/qt/file-tree.cc transmission-2.82/qt/file-tree.cc
--- transmission-2.82.orig/qt/file-tree.cc	2013-08-09 04:45:54.254416490 +0200
+++ transmission-2.82/qt/file-tree.cc	2013-09-07 15:14:06.993465096 +0200
@@ -885,7 +885,11 @@
   for (int i=0; i<NUM_COLUMNS; ++i)
     {
       setColumnHidden (i, (i<FIRST_VISIBLE_COLUMN) || (LAST_VISIBLE_COLUMN<i));
+#if QT_VERSION >= 0x050000
       header()->setSectionResizeMode(i, QHeaderView::Interactive);
+#else
+      header()->setResizeMode(i, QHeaderView::Interactive);
+#endif
     }
 
   connect (this, SIGNAL(clicked(const QModelIndex&)),
